# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: 2ed5e139d784
# Full Hash: 2ed5e139d7840692dad5bcf39348e92bf3b05fdb
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-12-05 09:38:58
# Regressor Bug: 1678030
# File Overlap Count: 1
# Description:
#   Bug 1678030 - Move CreateAndExecuteSingleStepStatement to QuotaCommon and use in all quota clients. r=dom-workers-and-storage-reviewers,janv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D97460
# ==============================================================================

diff -r 80442bbcf255 -r 2ed5e139d784 dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Fri Dec 04 15:08:17 2020 +0000
+++ b/dom/indexedDB/ActorsParent.cpp	Fri Dec 04 14:56:18 2020 +0000
@@ -685,20 +685,6 @@
   return NS_OK;
 }
 
-Result<nsCOMPtr<mozIStorageStatement>, nsresult>
-CreateAndExecuteSingleStepStatement(mozIStorageConnection& aConnection,
-                                    const nsACString& aStatementString) {
-  IDB_TRY_UNWRAP(auto stmt, MOZ_TO_RESULT_INVOKE_TYPED(
-                                nsCOMPtr<mozIStorageStatement>, aConnection,
-                                CreateStatement, aStatementString));
-
-  IDB_TRY_UNWRAP(const DebugOnly<bool> hasResult,
-                 MOZ_TO_RESULT_INVOKE(stmt, ExecuteStep));
-  MOZ_ASSERT(hasResult);
-
-  return stmt;
-}
-
 template <typename StepFunc>
 Result<Ok, nsresult> CollectWhileHasResult(mozIStorageStatement& aStmt,
                                            StepFunc&& aStepFunc) {